airflow/configuration.py

```diff
@@ -341,12 +381,15 @@ def get(self, section, key, **kwargs):
             return expand_env_var(d[section][key])
 
         else:
-            logging.warn("section/key [{section}/{key}] not found "
-                         "in config".format(**locals()))
+            if throwOnNonExist :
```

Comment:
Given that this is already a bunch of `elif` shall we just make this another `elif`?